src/module_2/groceries_analysis.md

```diff
@@ -0,0 +1,1811 @@
+plt.ylabel('Revenue')
+plt.show()
+
+# Calculate percentage of revenue from the top 10 items
+top_10_revenue = common_items['revenue'][:10].sum()
+total_revenue = common_items['revenue'].sum()
+top_10_revenue_percentage = top_10_revenue / total_revenue * 100
+printmd(f"Percentage of revenue from the top 10 items: {top_10_revenue_percentage:.2f}%")
+# Calculate percentage of revenue from the rest of the dataset
+rest_revenue = common_items['revenue'][10:].sum()
+rest_revenue_percentage = rest_revenue / total_revenue * 100
+printmd(f"Percentage of revenue from the rest of the dataset: {rest_revenue_percentage:.2f}%")
+# Calculate percentage of revenue from the top product 
+top_product_revenue = common_items['revenue'][0]
+top_product_revenue_percentage = top_product_revenue / total_revenue * 100
+printmd(f"Percentage of revenue from the top product: {top_product_revenue_percentage:.2f}%")
+
+```
+
+
+    
+![png](groceries_analysis_files/groceries_analysis_19_0.png)
+    
+
+
+
+Percentage of revenue from the top 10 items: 20.68%
+
+
+
+Percentage of revenue from the rest of the dataset: 79.32%
+
+
+
+Percentage of revenue from the top product: 9.88%
+
+
+The platform's product portfolio doesn't appear to be well balanced in terms of its item revenues, with the top ten products accounting for 20% of the total revenue, while the other 1467 products account for the remaining 80% . Moreover, the top product represents 10% of the total revenue, which is a lot for a single product. Let's now check revenue by general categories.
+
+
+```python
+category_revenue = common_items.groupby('product_type')['revenue'].sum().reset_index().sort_values('revenue', ascending=False)
+fig, ax = plt.subplots(1,2, figsize=(15,5))
+ax[0].bar(x=category_revenue['product_type'], height=category_revenue['revenue'], edgecolor='k')
+ax[0].set_title('Categories by Revenue')
+ax[0].set_xlabel('Category')
+ax[0].set_ylabel('Revenue')
+ax[0].set_xticklabels(category_revenue['product_type'],rotation=45, ha='right')
+
+
+ax[1].bar(x=category_revenue['product_type'][:10], height=category_revenue['revenue'][:10], edgecolor='k')
+ax[1].set_title('Top 10 Categories by Revenue')
+ax[1].set_xlabel('Category')
+ax[1].set_ylabel('Revenue')
+ax[1].set_xticklabels(category_revenue['product_type'][:10],rotation=45, ha='right')
+
+plt.tight_layout()
+plt.show()
+```
+
+    /tmp/ipykernel_4591/1866416911.py:7: UserWarning: set_ticklabels() should only be used with a fixed number of ticks, i.e. after set_ticks() or using a FixedLocator.
+      ax[0].set_xticklabels(category_revenue['product_type'],rotation=45, ha='right')
+    /tmp/ipykernel_4591/1866416911.py:14: UserWarning: set_ticklabels() should only be used with a fixed number of ticks, i.e. after set_ticks() or using a FixedLocator.
+      ax[1].set_xticklabels(category_revenue['product_type'][:10],rotation=45, ha='right')
+
+
+
+    
+![png](groceries_analysis_files/groceries_analysis_21_1.png)
+    
+
+
+Key Insights:
```

Comment:
I love it! Always, always generate conclusive cells with insights and action points